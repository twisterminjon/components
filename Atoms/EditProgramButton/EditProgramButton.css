.editprogrambutton-button {
  position: relative;
  background-color: transparent;
  color: var(--white);
  border-radius: 4px;
  border: none;
  outline: none;

  display: flex;
  align-items: center;
  justify-content: center;
}

.editprogrambutton-button:hover {
  background-color: var(--button_color_hover);
}

.editprogrambutton-button:active {
  background-color: var(--button_color_pressed);
}

.editprogrambutton-button:disabled,
.editprogrambutton-button[disabled] {
  color: var(--button_color_disabled);
  background-color: transparent;
}
