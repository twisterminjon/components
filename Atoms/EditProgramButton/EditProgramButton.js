import React from 'react'
import PropTypes from 'prop-types'
import './EditProgramButton.css'
import IconPencil from '../Icons/IconPencil'

// FIXME: Replace w/ ButtonIconSmall
export default function EditProgramButton({ onClick, disabled, className, ...rest }) {
  return (
    <button className={`editprogrambutton-button ${className}`} onClick={onClick} disabled={disabled} {...rest}>
      <IconPencil
        size={14}
        color={disabled ? 'var(--edit_icon_disabled_fg)' : 'var(--white)'}
        style={{ display: 'block' }}
      />
    </button>
  )
}

EditProgramButton.propTypes = {
  /** Function called when the button is clicked */
  onClick: PropTypes.func,
  /** If true, will show in a disabled this.state. */
  disabled: PropTypes.bool,
  /** ClassName for the wrapper */
  className: PropTypes.string,
}

EditProgramButton.defaultProps = {
  disabled: false,
  className: '',
}
