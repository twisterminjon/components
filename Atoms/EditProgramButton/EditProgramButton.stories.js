import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import EditProgramButton from './EditProgramButton'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('EditProgramButton', () => {
  const disabledLab = 'disabled'
  const disabledDef = false
  const disabledVal = boolean(disabledLab, disabledDef)

  return (
    <EditProgramButton
      onClick={() => {
        alert('clicked')
      }}
      disabled={disabledVal}
    />
  )
})
