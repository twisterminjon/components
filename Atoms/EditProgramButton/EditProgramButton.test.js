import React from 'react'
import EditProgramButton from './EditProgramButton'

describe('EditProgramButton', () => {
  const mockFun = jest.fn()

  it('EditProgramButton matches the snapshot', () => {
    const wrapper = window.shallow(<EditProgramButton onClick={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Uses the passed onClick function', () => {
    const wrapper = window.shallow(<EditProgramButton onClick={mockFun} />)
    wrapper.simulate('click')
    expect(mockFun).toHaveBeenCalled()
  })
})
