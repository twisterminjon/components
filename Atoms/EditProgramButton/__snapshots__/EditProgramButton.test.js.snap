// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProgramButton EditProgramButton matches the snapshot 1`] = `
<button
  className="editprogrambutton-button "
  disabled={false}
  onClick={[MockFunction]}
>
  <IconPencil
    color="var(--white)"
    size={14}
    style={
      Object {
        "display": "block",
      }
    }
  />
</button>
`;
